sycl/test/check_device_code/matrix/matrix-nvptx-single-bit-test.cpp

```diff
@@ -0,0 +1,71 @@
+// REQUIRES: cuda
+
+// RUN: %clangxx -fsycl-device-only -fsycl-targets=nvptx64-nvidia-cuda -Xsycl-target-backend --cuda-gpu-arch=sm_75 -DSYCL_EXT_ONEAPI_MATRIX=3 -S -Xclang -emit-llvm %s -o -| FileCheck %s
+
+#include <CL/sycl.hpp>
+
+using namespace sycl;
+using namespace sycl::ext::oneapi::experimental::matrix;
+
+// M, N, K define the sizes of dimensions of the three matrix types (a, b,
+// accumulator) used per subgroup operation.
+constexpr int M = 8;   // number of rows of accumulator,
+                       // number of cols of b.
+constexpr int N = 8;   // number of cols of accumulator,
+                       // number of rows of a.
+constexpr int K = 128; // number of cols of a/number of rows of b.
```

Comment:
you missed to make the change here. 
K should be 4 here.
Can you please add a comment where you are making these changes?
Basically, saying that the underlying intrinsics are expecting a shape of K equals to number of total bits, not number of elements.

Reply:
Thanks I forgot that.  Fixed now, and I also updated the test so it will work with the legacy pass-manager.

I've added a more detailed comment describing Bitwise Dot Product and how this dictates the relation between the number of Array elements used for A/B arrays and the number of single-bit matrix elements that the A/B arrays represent.  I've also correspondingly updated the test in intel/llvm-test-suite and the tensor cores matrix extension PR https://github.com/intel/llvm/pull/4695.